restcomm/restcomm.interpreter/src/test/java/org/restcomm/connect/interpreter/GatherSpeechTest.java

```diff
@@ -0,0 +1,400 @@
+                interpreter.tell(new MediaGroupResponse(new CollectedResult("Hello. World.", true, false)), observer);
+
+                callback = expectMsgClass(HttpRequestDescriptor.class);
+                assertEquals(callback.getUri(), actionCallbackUri);
+                assertEquals(findParam(callback.getParameters(), "SpeechResult").getValue(), "Hello. World.");
+
+                interpreter.tell(new DownloaderResponse(getOkRcml(partialCallbackUri, endRcml)), observer);
+
+                expectMsgClass(Hangup.class);
+            }
+        };
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testPartialAndPlayScenario() throws Exception {
+        new JavaTestKit(system) {
+            {
+                final ActorRef observer = getRef();
+                final ActorRef interpreter = createVoiceInterpreter(observer);
+                interpreter.tell(new StartInterpreter(observer), observer);
+
+                expectMsgClass(GetCallInfo.class);
+                interpreter.tell(new CallResponse(new CallInfo(
+                        new Sid("ACae6e420f425248d6a26948c17a9e2acf"),
+                        CallStateChanged.State.IN_PROGRESS,
+                        CreateCallType.SIP,
+                        "inbound",
+                        new DateTime(),
+                        null,
+                        "test", "test",
+                        "testTo",
+                        null,
+                        null,
+                        false,
+                        false,
+                        false,
+                        new DateTime())), observer);
+
+                expectMsgClass(Observe.class);
+
+                //wait for rcml downloading
+                HttpRequestDescriptor callback = expectMsgClass(HttpRequestDescriptor.class);
+                assertEquals(callback.getUri(), requestUri);
+
+                interpreter.tell(new DownloaderResponse(getOkRcml(requestUri, gatherRcml)), observer);
+
+                expectMsgClass(Collect.class);
+
+                //generate partial response1
+                interpreter.tell(new MediaGroupResponse(new CollectedResult("1", true, true)), observer);
+
+                callback = expectMsgClass(HttpRequestDescriptor.class);
+                assertEquals(callback.getUri(), partialCallbackUri);
+                assertEquals(findParam(callback.getParameters(), "UnstableSpeechResult").getValue(), "1");
+
+                interpreter.tell(new DownloaderResponse(getOkRcml(partialCallbackUri, "")), observer);
+
+                //generate partial response2
+                interpreter.tell(new MediaGroupResponse(new CollectedResult("12", true, true)), observer);
+
+                callback = expectMsgClass(HttpRequestDescriptor.class);
+                assertEquals(callback.getUri(), partialCallbackUri);
+                assertEquals(findParam(callback.getParameters(), "UnstableSpeechResult").getValue(), "12");
```

Comment:
Name of this parameter *UnstableSpeechResult* is confusing. we should make this *PartialSpeechResult*.